src/rqt_publisher/publisher.py

```diff
@@ -101,13 +101,21 @@ def _add_publisher(self, publisher_info):
         if publisher_info['message_instance'] is None:
             return
 
-        # create publisher and timer
+        # Following code taken from
+        # https://github.com/ros2/ros2cli/blob/master/ros2topic/ros2topic/verb/pub.py
+
         try:
-            publisher_info['publisher'] = rospy.Publisher(
-                publisher_info['topic_name'], type(publisher_info['message_instance']), queue_size=100)
-        except TypeError:
-            publisher_info['publisher'] = rospy.Publisher(
-                publisher_info['topic_name'], type(publisher_info['message_instance']))
+            package_name, message_name = publisher_info['type_name'].split('/', 2)
+            if not package_name or not message_name:
+                raise ValueError()
+        except ValueError:
+            raise RuntimeError('The passed message type is invalid')
+        module = importlib.import_module(package_name + '.msg')
```

Comment:
Won't this need try except in case the package name doesn't have a msg?

Reply:
This logic is already wrapped in a try catch. So any errors during 'add_publisher' will be caught